materials/for-loops-R.md

```diff
@@ -208,140 +201,120 @@ waterbirds_cap_list
+* Loops go over columns of dataframes
 
 ```
-pets <- data.frame(pet_name = c("spot", "gigantor", "fluffy"),
-                   pet_type = c("fish", "hamster", "lizard"))
-for (pet in pets){
+waterbirds <- data.frame(sci_name = c("cygnus olor", 
+                                      "aix sponsa", 
+                                      "anas acuta"), 
+                         common_name = c("mute swan", 
+                                         "wood duck", 
+                                         "pintail"))
+for (waterbird in waterbirds){
   print("Start new loop")
-  print(pet)
+  print(waterbird)
 }
 ```
 
-* Who has seen/used a different approach to looping than has been show?
-* That's called looping over an `index` and while it's overly complicated for
-  simple loops it's useful for more complicated things like looping over rows
+* Can loop over rows of data frames using index
 
 ```
-pets <- data.frame(pet_name = c("spot", "gigantor", "fluffy"),
-                   pet_type = c("fish", "hamster", "lizard"))
-for (i in 1:nrow(pets)){
-  print(paste(pets$pet_name[i], " is a ", pets$pet_type[i], sep =""))
+for (i in 1:nrow(waterbirds)){
+  print(i)
 }
 ```
 
-* We can also use the index to store the output by pre-allocating memory
-* This is a lot faster than adding rows because it doesn't copy the data frame
-  every time through the loop
+* Index can be any letter/word, i is convention
 
 ```
-output <- data.frame(name = character(3), namelength = numeric(3),
-                     stringsAsFactors = FALSE)
-for (i in 1:nrow(pets)) {
-  pet_upper <- str_to_upper(pets$pet_name[i])
-  pet_length <- str_length(pets$pet_type[i])
-  output[i,] <- c(pet_upper, pet_length)
+for (r in 1:nrow(waterbirds)){
+  print(r)
 }
 ```
 
-> Assign [Exercise 4 - stringr]({{ site.baseurl }}/exercises/Loops-stringr-R).
-> Assign [Exercise 5 - DNA or RNA]({{ site.baseurl }}/exercises/Making-choices-dna-or-rna-R).
-
-
-### Alternatives to loops
-
-#### Vectorization
-
-* Operations that work on all of the elements in a vector
-* Make functions that work with vectors so no need to loop over values
-
 ```
-lengths <- c(2.26, 1.48, 3.84)
-mass <- 0.73 * length ** 3.63
-mass
+for (i in 1:nrow(waterbirds)){
+  print(waterbirds$sci_name[i])
+}
 ```
 
-* Works for most simple math
-
-> Do [for Loop]({{ site.baseurl }}/exercises/Functions-for-loop-R).
-
-#### Looping over files
-
-* To learn alternatives learn how to do the same thing with many files. First
-  with loops and then with some alternatives in R
-* E.g., we have a set of files with satellite collar location data
-
 ```
-date,time,lat,long
-2016-01-01,4:20,26.16,-35.28
+for (i in 1:nrow(waterbirds)){
+  print(paste(waterbirds$sci_name[i], "is a", 
+              waterbirds$common_name[i]))
+}
 ```
 
-* To get a list of all these files in our `data` directory use `list.files()`
+* Less memory to create initial empty dataframe
+* Creates copy of dataframe when adding rows
```

Comment:
This might also be too complicated for day 1 loops. Good responses to have available for questions though.